src/components/ShelterListItem/ShelterListItem.tsx

```diff
@@ -22,13 +22,20 @@ const ShelterListItem = (props: IShelterListItemProps) => {
     );
 
   const tags = useMemo(
-    () =>
-      data.shelterSupplies
-        .filter((s) => s.priority >= SupplyPriority.Needing)
-        .sort((a, b) => b.priority - a.priority),
+    () => {
+      return data.shelterSupplies?.filter((s) => !getCategoriesToFilterVolunteers().some(c => c.includes(s.supply?.supplyCategory?.name.toLowerCase())))
+        .sort((a, b) => b.priority - a.priority).slice(0, 10)
+    },
```

Comment:
Esse trecho de código está repetido na linha 34. Talvez seja legal extrair ele para outra variável e usar nos dois lugares fazendo as alterações necessárias:

```suggestion
    const shelterSuppliesFiltered = useMemo(() => {
      return data.shelterSupplies?.filter((s) => !getCategoriesToFilterVolunteers().some(
        c => c.includes(s.supply?.supplyCategory?.name.toLowerCase()))
    )},
    [data.shelterSupplies]
  );
  
  const tags = shelterSuppliesFiltered.sort((a, b) => b.priority - a.priority).slice(0, 10);
  
  const volunteerTags =  shelterSuppliesFiltered.reverse();
```